### PR TITLE
babel 7 compliance. Add optional parameter babelCore.

### PR DESCRIPTION
**Pull request by SSANSH**

### CODE REVIEW

**Comment by SSANSH** [comment]
@zyrorl, do you have comment about this pull request?

File: src/transforms/Babel.ts [hunk 1]
```diff
@@ -21,19 +22,27 @@ export class BabelTransform {
...
   run ({ code, map, relPath }) {
     let result = { code, map, relPath };
 
     try {
-      const transformed = this.babel.transform(code, {
-        ...this.config,
-        sourceFileName  : relPath,
-        sourceMapTarget : relPath,
-      });
+      let transformed = null;
```
Comment (nfour) [inline src/transforms/Babel.ts]: Write this like

```ts
		const sourceMapTarget = semver.gt(... blah blah) ? undefined : relPath;
        const transformed = this.babel.transform(code, {
          ...this.config,
          sourceFileName: relPath,
          sourceMapTarget,
        });

```

File: src/SourceBundler.ts [hunk 2]
```diff
@@ -28,6 +29,7 @@ export class SourceBundler {
     archive: Archiver;
     servicePath: string;
     babel?: any;
+    babelCore?: any;
```
Comment (nfour) [inline src/SourceBundler.ts]: Sounds like its a `string`

File: src/ServerlessBuildPlugin.ts [hunk 3]
```diff
@@ -292,6 +292,7 @@ export class ServerlessBuildPlugin {
           ? this.config.uglify
           : undefined,
         babel: this.config.babel,
+        babelCore: this.config.babelCore ? this.config.babelCore : 'babel-core',
```
Comment (nfour) [inline src/ServerlessBuildPlugin.ts]: `babelCore: this.config.babelCore || 'babel-core'` is more terse

File: docs/Install & Config.md [hunk 4]
```diff
@@ -50,6 +50,10 @@ method: bundle
 #  Default is `null`.
 babel: true
 
+# Define core babel package
```
Comment (nfour) [inline docs/Install & Config.md]: Describe why youd ever want to specify this option. Its not clear from this MR at all why you'd need to do this.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,5 @@
+## [Unreleased]
+- babel 7 compliance. Add optional parameter babelCore.
```
Comment (nfour) [inline CHANGELOG.md]: Why?

**Review by nfour — CHANGES_REQUESTED** [review]

**Comment by SSANSH** [comment]
Hi, I applied changes and test change requested by nfour (babel6, babel7)
Thanks.

File: src/transforms/Babel.ts (hunk 1 above)
Comment (nfour) [reply src/transforms/Babel.ts]: Dude I gave you the code, just copy paste it, dont `delete` that is RETARDED

Comment (SSANSH) [reply src/transforms/Babel.ts]: @nfour your code don't works, if I applied then, I already have the field sourceMapTarget = undefined and so babel7 tell me that the field is not compliant with babel7. thats why I recode this with a delete which is the second fashion to remove a property into an object on nodejs.
Thanks

Comment (nfour) [reply src/transforms/Babel.ts]: Then define a subset of the object in the ternary instead, then spread it in eg.

```ts
const transformConfig = semver.gt(... blah blah) ? { ... }: { ... ;
const transformed = (blah blah) {
  ...this.config,
 ...transformConfig,
}
```

:)

**Comment by SSANSH** [comment]
Hi @nfour, change applied following your requirement. tested with babel6 / babel7.

File: src/transforms/Babel.ts (hunk 1 above)
Comment (SSANSH) [reply src/transforms/Babel.ts]: @nfour resolved.

**Review by nfour — APPROVED** [review]

**Comment by nfour** [comment]
Looks good

**Comment by nfour** [comment]
3.2.0 published

**Comment by SSANSH** [comment]
@nfour thanks for the merge I used latest package publish with success.

**Pull request merged**